command_node.py

```diff
@@ -0,0 +1,50 @@
+# Represents a node in a tree, this node knows its parent
+class CommandNode:
+    def __init__(self,
+                 tmp_name,
+                 tmp_parent=None,
+                 tmp_parameter_list=list(),
+                 tmp_command_info="Currently none available"):
+
+        assert (type(tmp_name) is not None)
+        self.name = tmp_name.upper()
+        self.children = dict()
+        self.parent = tmp_parent
+        self.parameter_list = tmp_parameter_list
+        self.command_info = tmp_command_info
+
+    # This function adds a child, or overrides it when it exists
+    def add_child(self, tmp_child):
+        self.children[tmp_child.name.upper()] = tmp_child
+
+    # Gets all parents in a list, ordered as root first
+    def get_all_parents(self):
+        if self.parent is None:
+            return [self.name]
+        else:
+            return self.parent.get_all_parents() + [self.name]
+
+    # Sets the parent to given parent
+    def set_parent(self, tmp_parent):
+        self.parent = tmp_parent
+
+    # Displays the children of this node
+    def show_children(self):
+        if len(self.children) > 0:
+            for child_key in self.children:
+                print(self.children[child_key].name)
+
+    # Returns true if this node has give key (name) as a direct child, else returns false
+    def has_child(self, key):
+        return key.upper() in self.children
+
+    # Gets the amount of children this node has
+    def get_child_amount(self):
+        return len(self.children)
+
+    # Returns a child at the given key. Returns None if no key matches.
+    def get_child(self, key):
+        if key.upper() in self.children:
+            return self.children[key.upper()]
+        else:
+            return None
```

Comment:
I had a conversation with Jens about this, we agreed that the best way to handle this is to take on the behaviour of `dict`:
 - `get` will return `None` if the key is not defined (because the default value `get` returns is `None`, `dict.get(key, default = None)`)
 - `[]` will raise a `KeyError` when the key is not present

In this case, `get_child` can be refactored to the following:
```python
def get_child(self, key):
    return self.children.get(key.upper())
```

Reply:
that seems resonable,

i think this is an additional argument for making CommandNode inherit from dict.
then other peaple can choose what functionality they want:
KeyError on missing key: dict.__getitem__(CommandNode(), key) or CommandNode()[key]
return None on missing key: dict.get(CommandNode(), key) or CommandNode().get(key)